docs/security.rst

```diff
@@ -1,87 +1,146 @@
-`RFC 7662 <rfc7662_>`_. Note that you have to validate the ``active``
-or ``exp`` fields etc. yourself.
+While the validation functions should accept different arguments based on the authentication type
+(as documented below), they should all return a dict which complies with `RFC 7662 <rfc7662_>`_:
 
-The Token Info response will be passed in the ``token_info`` argument to the handler
-function. The ``sub`` property of the Token Info response will be passed in the ``user``
-argument to the handler function.
+.. code-block:: json
 
-Deprecated features, retained for backward compatibility:
+    {
+      "active": true,
+      "client_id": "l238j323ds-23ij4",
+      "username": "jdoe",
+      "scope": "read write dolphin",
+      "sub": "Z5O3upPC88QrAjx00dis",
+      "aud": "https://protected.example.net/resource",
+      "iss": "https://server.example.com/",
+      "exp": 1419356238,
+      "iat": 1419350238,
+      "extension_field": "twenty-seven"
+    }
 
-- As alternative to ``x-tokenInfoFunc``, you can set ``x-tokenInfoUrl`` or
-  ``TOKENINFO_URL`` env var. It must contain a URL to validate and get the token
-  information which complies with `RFC 6749 <rfc6749_>`_.
-  When both ``x-tokenInfoUrl`` and ``x-tokenInfoFunc`` are used, Connexion
-  will prioritize the function method. Connexion expects the authorization
-  server to receive the OAuth token in the ``Authorization`` header field in the
-  format described in `RFC 6750 <rfc6750_>`_ section 2.1. This aspect represents
-  a significant difference from the usual OAuth flow.
-- ``scope`` field can also be named ``scopes``.
-- ``sub`` field can also be named ``uid``.
+The token information is made available to your endpoint view functions via the
+:ref:`context <context:context.context>`, which you can also have passed in as an
+:ref:`argument <request:Context>`.
 
-You can find a `minimal OAuth example application`_ showing the use of
-``x-tokenInfoUrl``, and `another OAuth example`_ showing the use of
-``x-tokenInfoFunc`` in Connexion's "examples" folder.
+.. note::
 
-.. _minimal OAuth example application: https://github.com/spec-first/connexion/tree/main/examples/oauth2
-.. _another OAuth example: https://github.com/spec-first/connexion/tree/main/examples/oauth2_local_tokeninfo
+    Note that you are responsible to validate any fields other than the scopes yourself.
+
+.. _rfc7662: https://tools.ietf.org/html/rfc7662
 
 Basic Authentication
 --------------------
 
-With Connexion, the API security definition **must** include a
-``x-basicInfoFunc`` or set ``BASICINFO_FUNC`` env var. It uses the same
-semantics as for ``x-tokenInfoFunc``, but the function accepts three
-parameters: username, password and required_scopes.
+For Basic authentication, the API security definition must include an
+``x-basicInfoFunc`` definition or set the ``BASICINFO_FUNC`` environment variable.
+
+The function should accept the following arguments:
+- username
```

Comment:
These don't get rendered properly:
<img width="517" alt="image" src="https://github.com/spec-first/connexion/assets/33490966/19fac76c-d07f-43c7-8705-4b0cbddc8545">

I believe an extra empty line is needed before the first list item